services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -267,16 +256,13 @@ On update following will happen automatically:
                 )
             ]
     )
+    @PreAuthorize("@hankeAuthorizer.authorizeHankeTunnus(#hankeTunnus, 'DELETE')")
     fun deleteHanke(@PathVariable hankeTunnus: String) {
         logger.info { "Deleting hanke: $hankeTunnus" }
 
+        // TODO: Move getHankeWithApplications inside deleteHanke
```

Comment:
This is a good idea, but I recommend writing a jira task instead of todo comments in the code.

Reply:
I did the refactoring in ea7678ad7ddca7d7db1ea3c3c0011ffea9b3ed38.

I also removed HankeWithApplications completely since it wasn't really needed anywhere.